src/main/java/seedu/tp/parser/Parser.java

```diff
@@ -63,9 +65,15 @@ public Command parseCommand(String userInput) throws HistoryFlashcardException {
             return new OtherFlashcardCommand(flashcardList, flashcardFactory);
         case LIST_COMMAND:
             return new ListCommand(flashcardList, ui);
+        case SHOW_COMMAND:
+            try {
+                return new ShowCommand(flashcardList, Integer.parseInt(splitInput[1]), ui);
+            } catch (Exception e) {
```

Comment:
I think it's generally a better idea to be explicit about what Exception we are catching, right?

Reply:
Noted and changed to IndexOutOfBoundsException.